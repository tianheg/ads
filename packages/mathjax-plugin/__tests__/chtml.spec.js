/** @jest-environment node */

const { createApp, testCases } = require('./utils')

describe('mathjax-chtml', () => {
  let app

  beforeAll(async () => {
    app = createApp('chtml')
    return app.process()
  })

  testCases.forEach(({ name, content, frontmatter }) => {
    test(name, () => {
      const { html } = app.markdown.render(content, { frontmatter })
      expect(html).toMatchSnapshot()
    })
  })
})
