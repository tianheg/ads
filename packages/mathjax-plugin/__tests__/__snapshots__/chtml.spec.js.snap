// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mathjax-chtml block.md 1`] = `
<p>
  <p>
    <mjx-container class="MathJax" jax="CHTML" display="true">
      <mjx-math display="true" class=" MJX-TEX">
        <mjx-mfrac>
          <mjx-frac type="d">
            <mjx-num>
              <mjx-nstrut type="d"></mjx-nstrut>
              <mjx-msup>
                <mjx-mi class="mjx-n">
                  <mjx-c c="2202"></mjx-c>
                </mjx-mi>
                <mjx-script style="vertical-align: 0.363em">
                  <mjx-mi class="mjx-i" size="s">
                    <mjx-c c="r"></mjx-c>
                  </mjx-mi>
                </mjx-script>
              </mjx-msup>
            </mjx-num>
            <mjx-dbox>
              <mjx-dtable>
                <mjx-line type="d"></mjx-line>
                <mjx-row>
                  <mjx-den>
                    <mjx-dstrut type="d"></mjx-dstrut>
                    <mjx-mrow>
                      <mjx-mi class="mjx-n">
                        <mjx-c c="2202"></mjx-c>
                      </mjx-mi>
                      <mjx-msup>
                        <mjx-mi class="mjx-i">
                          <mjx-c c="3C9"></mjx-c>
                        </mjx-mi>
                        <mjx-script style="vertical-align: 0.363em">
                          <mjx-mi class="mjx-i" size="s">
                            <mjx-c c="r"></mjx-c>
                          </mjx-mi>
                        </mjx-script>
                      </mjx-msup>
                    </mjx-mrow>
                  </mjx-den>
                </mjx-row>
              </mjx-dtable>
            </mjx-dbox>
          </mjx-frac>
        </mjx-mfrac>
        <mjx-mrow>
          <mjx-mo class="mjx-s3">
            <mjx-c c="("></mjx-c>
          </mjx-mo>
          <mjx-mfrac>
            <mjx-frac type="d">
              <mjx-num>
                <mjx-nstrut type="d"></mjx-nstrut>
                <mjx-msup>
                  <mjx-mi class="mjx-i">
                    <mjx-c c="y"></mjx-c>
                  </mjx-mi>
                  <mjx-script style="vertical-align: 0.363em">
                    <mjx-TeXAtom size="s">
                      <mjx-mi class="mjx-i">
                        <mjx-c c="3C9"></mjx-c>
                      </mjx-mi>
                    </mjx-TeXAtom>
                  </mjx-script>
                </mjx-msup>
              </mjx-num>
              <mjx-dbox>
                <mjx-dtable>
                  <mjx-line type="d"></mjx-line>
                  <mjx-row>
                    <mjx-den>
                      <mjx-dstrut type="d"></mjx-dstrut>
                      <mjx-mi class="mjx-i">
                        <mjx-c c="3C9"></mjx-c>
                      </mjx-mi>
                    </mjx-den>
                  </mjx-row>
                </mjx-dtable>
              </mjx-dbox>
            </mjx-frac>
          </mjx-mfrac>
          <mjx-mo class="mjx-s3">
            <mjx-c c=")"></mjx-c>
          </mjx-mo>
        </mjx-mrow>
        <mjx-mo class="mjx-n" space="4">
          <mjx-c c="="></mjx-c>
        </mjx-mo>
        <mjx-mrow space="4">
          <mjx-mo class="mjx-s3">
            <mjx-c c="("></mjx-c>
          </mjx-mo>
          <mjx-mfrac>
            <mjx-frac type="d">
              <mjx-num>
                <mjx-nstrut type="d"></mjx-nstrut>
                <mjx-msup>
                  <mjx-mi class="mjx-i">
                    <mjx-c c="y"></mjx-c>
                  </mjx-mi>
                  <mjx-script style="vertical-align: 0.363em">
                    <mjx-TeXAtom size="s">
                      <mjx-mi class="mjx-i">
                        <mjx-c c="3C9"></mjx-c>
                      </mjx-mi>
                    </mjx-TeXAtom>
                  </mjx-script>
                </mjx-msup>
              </mjx-num>
              <mjx-dbox>
                <mjx-dtable>
                  <mjx-line type="d"></mjx-line>
                  <mjx-row>
                    <mjx-den>
                      <mjx-dstrut type="d"></mjx-dstrut>
                      <mjx-mi class="mjx-i">
                        <mjx-c c="3C9"></mjx-c>
                      </mjx-mi>
                    </mjx-den>
                  </mjx-row>
                </mjx-dtable>
              </mjx-dbox>
            </mjx-frac>
          </mjx-mfrac>
          <mjx-mo class="mjx-s3">
            <mjx-c c=")"></mjx-c>
          </mjx-mo>
        </mjx-mrow>
        <mjx-mrow>
          <mjx-mo class="mjx-s4">
            <mjx-c c="{"></mjx-c>
          </mjx-mo>
          <mjx-mo class="mjx-n">
            <mjx-c c="("></mjx-c>
          </mjx-mo>
          <mjx-mi class="mjx-n">
            <mjx-c c="l"></mjx-c>
            <mjx-c c="o"></mjx-c>
            <mjx-c c="g"></mjx-c>
          </mjx-mi>
          <mjx-mo class="mjx-n">
            <mjx-c c="2061"></mjx-c>
          </mjx-mo>
          <mjx-mi class="mjx-i" space="2">
            <mjx-c c="y"></mjx-c>
          </mjx-mi>
          <mjx-msup>
            <mjx-mo class="mjx-n">
              <mjx-c c=")"></mjx-c>
            </mjx-mo>
            <mjx-script style="vertical-align: 0.413em">
              <mjx-mi class="mjx-i" size="s">
                <mjx-c c="r"></mjx-c>
              </mjx-mi>
            </mjx-script>
          </mjx-msup>
          <mjx-mo class="mjx-n" space="3">
            <mjx-c c="+"></mjx-c>
          </mjx-mo>
          <mjx-munderover space="3">
            <mjx-over style="padding-bottom: 0.192em; padding-left: 0.563em">
              <mjx-mi class="mjx-i" size="s">
                <mjx-c c="r"></mjx-c>
              </mjx-mi>
            </mjx-over>
            <mjx-box>
              <mjx-munder>
                <mjx-row>
                  <mjx-base>
                    <mjx-mo class="mjx-lop">
                      <mjx-c c="2211"></mjx-c>
                    </mjx-mo>
                  </mjx-base>
                </mjx-row>
                <mjx-row>
                  <mjx-under style="padding-top: 0.167em; padding-left: 0.148em">
                    <mjx-TeXAtom size="s">
                      <mjx-mi class="mjx-i">
                        <mjx-c c="i"></mjx-c>
                      </mjx-mi>
                      <mjx-mo class="mjx-n">
                        <mjx-c c="="></mjx-c>
                      </mjx-mo>
                      <mjx-mn class="mjx-n">
                        <mjx-c c="1"></mjx-c>
                      </mjx-mn>
                    </mjx-TeXAtom>
                  </mjx-under>
                </mjx-row>
              </mjx-munder>
            </mjx-box>
          </mjx-munderover>
          <mjx-mfrac space="2">
            <mjx-frac type="d">
              <mjx-num>
                <mjx-nstrut type="d"></mjx-nstrut>
                <mjx-mrow>
                  <mjx-mo class="mjx-n">
                    <mjx-c c="("></mjx-c>
                  </mjx-mo>
                  <mjx-mo class="mjx-n">
                    <mjx-c c="2212"></mjx-c>
                  </mjx-mo>
                  <mjx-mn class="mjx-n">
                    <mjx-c c="1"></mjx-c>
                  </mjx-mn>
                  <mjx-msup>
                    <mjx-mo class="mjx-n">
                      <mjx-c c=")"></mjx-c>
                    </mjx-mo>
                    <mjx-script style="vertical-align: 0.363em">
                      <mjx-mi class="mjx-i" size="s">
                        <mjx-c c="i"></mjx-c>
                      </mjx-mi>
                    </mjx-script>
                  </mjx-msup>
                  <mjx-mi class="mjx-i">
                    <mjx-c c="r"></mjx-c>
                  </mjx-mi>
                  <mjx-mo class="mjx-n" space="2">
                    <mjx-c c="22EF"></mjx-c>
                  </mjx-mo>
                  <mjx-mo class="mjx-n" space="2">
                    <mjx-c c="("></mjx-c>
                  </mjx-mo>
                  <mjx-mi class="mjx-i">
                    <mjx-c c="r"></mjx-c>
                  </mjx-mi>
                  <mjx-mo class="mjx-n" space="3">
                    <mjx-c c="2212"></mjx-c>
                  </mjx-mo>
                  <mjx-mi class="mjx-i" space="3">
                    <mjx-c c="i"></mjx-c>
                  </mjx-mi>
                  <mjx-mo class="mjx-n" space="3">
                    <mjx-c c="+"></mjx-c>
                  </mjx-mo>
                  <mjx-mn class="mjx-n" space="3">
                    <mjx-c c="1"></mjx-c>
                  </mjx-mn>
                  <mjx-mo class="mjx-n">
                    <mjx-c c=")"></mjx-c>
                  </mjx-mo>
                  <mjx-mo class="mjx-n">
                    <mjx-c c="("></mjx-c>
                  </mjx-mo>
                  <mjx-mi class="mjx-n">
                    <mjx-c c="l"></mjx-c>
                    <mjx-c c="o"></mjx-c>
                    <mjx-c c="g"></mjx-c>
                  </mjx-mi>
                  <mjx-mo class="mjx-n">
                    <mjx-c c="2061"></mjx-c>
                  </mjx-mo>
                  <mjx-mi class="mjx-i" space="2">
                    <mjx-c c="y"></mjx-c>
                  </mjx-mi>
                  <mjx-msup>
                    <mjx-mo class="mjx-n">
                      <mjx-c c=")"></mjx-c>
                    </mjx-mo>
                    <mjx-script style="vertical-align: 0.363em">
                      <mjx-TeXAtom size="s">
                        <mjx-mi class="mjx-i">
                          <mjx-c c="r"></mjx-c>
                        </mjx-mi>
                        <mjx-mo class="mjx-n">
                          <mjx-c c="2212"></mjx-c>
                        </mjx-mo>
                        <mjx-mi class="mjx-i">
                          <mjx-c c="i"></mjx-c>
                        </mjx-mi>
                      </mjx-TeXAtom>
                    </mjx-script>
                  </mjx-msup>
                </mjx-mrow>
              </mjx-num>
              <mjx-dbox>
                <mjx-dtable>
                  <mjx-line type="d"></mjx-line>
                  <mjx-row>
                    <mjx-den>
                      <mjx-dstrut type="d"></mjx-dstrut>
                      <mjx-msup>
                        <mjx-mi class="mjx-i">
                          <mjx-c c="3C9"></mjx-c>
                        </mjx-mi>
                        <mjx-script style="vertical-align: 0.363em">
                          <mjx-mi class="mjx-i" size="s">
                            <mjx-c c="i"></mjx-c>
                          </mjx-mi>
                        </mjx-script>
                      </mjx-msup>
                    </mjx-den>
                  </mjx-row>
                </mjx-dtable>
              </mjx-dbox>
            </mjx-frac>
          </mjx-mfrac>
          <mjx-mo class="mjx-s4">
            <mjx-c c="}"></mjx-c>
          </mjx-mo>
        </mjx-mrow>
      </mjx-math>
    </mjx-container>
  </p>
</p>
`;

exports[`mathjax-chtml convention.md 1`] = `<p>Anything between two $ characters will be treated as TeX math. The opening $ must have a non-space character immediately to its right, while the closing $ must have a non-space character immediately to its left, and must not be followed immediately by a digit. Thus, $20,000 and $30,000 won’t parse as math. If for some reason you need to enclose text in literal $ characters, backslash-escape them and they won’t be treated as math delimiters.</p>`;

exports[`mathjax-chtml inline.md 1`] = `
<p>Euler's identity <mjx-container class="MathJax" jax="CHTML">
    <mjx-math class=" MJX-TEX">
      <mjx-msup>
        <mjx-mi class="mjx-i">
          <mjx-c c="e"></mjx-c>
        </mjx-mi>
        <mjx-script style="vertical-align: 0.363em">
          <mjx-TeXAtom size="s">
            <mjx-mi class="mjx-i">
              <mjx-c c="i"></mjx-c>
            </mjx-mi>
            <mjx-mi class="mjx-i">
              <mjx-c c="3C0"></mjx-c>
            </mjx-mi>
          </mjx-TeXAtom>
        </mjx-script>
      </mjx-msup>
      <mjx-mo class="mjx-n" space="3">
        <mjx-c c="+"></mjx-c>
      </mjx-mo>
      <mjx-mn class="mjx-n" space="3">
        <mjx-c c="1"></mjx-c>
      </mjx-mn>
      <mjx-mo class="mjx-n" space="4">
        <mjx-c c="="></mjx-c>
      </mjx-mo>
      <mjx-mn class="mjx-n" space="4">
        <mjx-c c="0"></mjx-c>
      </mjx-mn>
    </mjx-math>
  </mjx-container> is a beautiful formula in <mjx-container class="MathJax" jax="CHTML">
    <mjx-math class=" MJX-TEX">
      <mjx-msup>
        <mjx-TeXAtom>
          <mjx-mi class="mjx-ds">
            <mjx-c c="R"></mjx-c>
          </mjx-mi>
        </mjx-TeXAtom>
        <mjx-script style="vertical-align: 0.41em">
          <mjx-mn class="mjx-n" size="s">
            <mjx-c c="2"></mjx-c>
          </mjx-mn>
        </mjx-script>
      </mjx-msup>
    </mjx-math>
  </mjx-container>.</p>
`;

exports[`mathjax-chtml macro.md 1`] = `
<p>We have <mjx-container class="MathJax" jax="CHTML">
    <mjx-math class=" MJX-TEX">
      <mjx-mi class="mjx-i">
        <mjx-c c="a"></mjx-c>
      </mjx-mi>
      <mjx-mo class="mjx-n" space="4">
        <mjx-c c=">"></mjx-c>
      </mjx-mo>
      <mjx-mi class="mjx-i" space="4">
        <mjx-c c="n"></mjx-c>
      </mjx-mi>
      <mjx-mo class="mjx-n" space="4">
        <mjx-c c="21D4"></mjx-c>
      </mjx-mo>
      <mjx-mi class="mjx-i" space="4">
        <mjx-c c="a"></mjx-c>
      </mjx-mi>
      <mjx-mo class="mjx-n" space="4">
        <mjx-c c="2A7E"></mjx-c>
      </mjx-mo>
      <mjx-mi class="mjx-i" space="4">
        <mjx-c c="n"></mjx-c>
      </mjx-mi>
      <mjx-mo class="mjx-n" space="3">
        <mjx-c c="+"></mjx-c>
      </mjx-mo>
      <mjx-mn class="mjx-n" space="3">
        <mjx-c c="1"></mjx-c>
      </mjx-mn>
    </mjx-math>
  </mjx-container>, if <mjx-container class="MathJax" jax="CHTML">
    <mjx-math class=" MJX-TEX">
      <mjx-mi class="mjx-i">
        <mjx-c c="a"></mjx-c>
      </mjx-mi>
      <mjx-mo class="mjx-n">
        <mjx-c c=","></mjx-c>
      </mjx-mo>
      <mjx-mi class="mjx-i" space="2">
        <mjx-c c="n"></mjx-c>
      </mjx-mi>
      <mjx-mo class="mjx-n" space="4">
        <mjx-c c="2208"></mjx-c>
      </mjx-mo>
      <mjx-TeXAtom space="4">
        <mjx-mi class="mjx-ds">
          <mjx-c c="Z"></mjx-c>
        </mjx-mi>
      </mjx-TeXAtom>
    </mjx-math>
  </mjx-container>.</p>
`;

exports[`mathjax-chtml preset.md 1`] = `
<p>
  <p>
    <mjx-container class="MathJax" jax="CHTML" display="true">
      <mjx-math display="true" class=" MJX-TEX">
        <mjx-mfrac>
          <mjx-frac type="d">
            <mjx-num>
              <mjx-nstrut type="d"></mjx-nstrut>
              <mjx-mi class="mjx-n">
                <mjx-c c="2202"></mjx-c>
              </mjx-mi>
            </mjx-num>
            <mjx-dbox>
              <mjx-dtable>
                <mjx-line type="d"></mjx-line>
                <mjx-row>
                  <mjx-den>
                    <mjx-dstrut type="d"></mjx-dstrut>
                    <mjx-mrow>
                      <mjx-mi class="mjx-n">
                        <mjx-c c="2202"></mjx-c>
                      </mjx-mi>
                      <mjx-mi class="mjx-i">
                        <mjx-c c="t"></mjx-c>
                      </mjx-mi>
                    </mjx-mrow>
                  </mjx-den>
                </mjx-row>
              </mjx-dtable>
            </mjx-dbox>
          </mjx-frac>
        </mjx-mfrac>
        <mjx-mrow>
          <mjx-mo class="mjx-s3">
            <mjx-c c="("></mjx-c>
          </mjx-mo>
          <mjx-mfrac>
            <mjx-frac type="d">
              <mjx-num>
                <mjx-nstrut type="d"></mjx-nstrut>
                <mjx-msup>
                  <mjx-mi class="mjx-i">
                    <mjx-c c="y"></mjx-c>
                  </mjx-mi>
                  <mjx-script style="vertical-align: 0.363em">
                    <mjx-mi class="mjx-i" size="s">
                      <mjx-c c="t"></mjx-c>
                    </mjx-mi>
                  </mjx-script>
                </mjx-msup>
              </mjx-num>
              <mjx-dbox>
                <mjx-dtable>
                  <mjx-line type="d"></mjx-line>
                  <mjx-row>
                    <mjx-den>
                      <mjx-dstrut type="d"></mjx-dstrut>
                      <mjx-mi class="mjx-i">
                        <mjx-c c="t"></mjx-c>
                      </mjx-mi>
                    </mjx-den>
                  </mjx-row>
                </mjx-dtable>
              </mjx-dbox>
            </mjx-frac>
          </mjx-mfrac>
          <mjx-mo class="mjx-s3">
            <mjx-c c=")"></mjx-c>
          </mjx-mo>
        </mjx-mrow>
      </mjx-math>
    </mjx-container>
  </p>
</p>
`;
